tsdb/shard.go

```diff
@@ -294,6 +294,10 @@ func (s *Shard) deleteMeasurement(name string, seriesKeys []string) error {
 		return err
 	}
 
+	// Remove entry from shard index.
+	s.mu.Lock()
```

Comment:
It doesn't seem necessary to lock for the entire function, so just doing it here.